### PR TITLE
Surface toggle trace event to the controller

### PR DESCRIPTION
**Pull request by microsoftsam**
Surface toggle trace event to the controller. 
- optional intervalSpeed parameter to alter execution trace speed.

### CODE REVIEW

File: webapp/src/app.tsx [hunk 1]
```diff
@@ -1013,13 +1013,13 @@ export class ProjectView
...
 
-    toggleTrace() {
+    toggleTrace(intervalSpeed?: number) {
         if (this.state.tracing) {
             this.editor.clearHighlightedStatements();
             simulator.setTraceInterval(0);
         }
         else {
-            simulator.setTraceInterval(simulator.SLOW_TRACE_INTERVAL);
+            simulator.setTraceInterval(intervalSpeed || simulator.SLOW_TRACE_INTERVAL);
```
Comment (riknoll) [inline webapp/src/app.tsx]: Is 0 valid?

Comment (microsoftsam) [inline webapp/src/app.tsx]: It seems to be the case, going off line 1019.

Comment (riknoll) [inline webapp/src/app.tsx]: Maybe change the check to `intervalSpeed == undefined`

Comment (microsoftsam) [inline webapp/src/app.tsx]: Done.

**Pull request merged**